Build fails on recent JDK11 versions

CI started reporting errors about javadocs.
I can reproduce the same error on JDK11 and JDK22.
Even upgrading the javadoc plugin does not help.
I will try to send a fix soon

```
[INFO] Total time:  01:43 min
[INFO] Finished at: 2019-07-25T21:41:14Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.0.0:jar (attach-javadocs) on project bookkeeper-stats-api: MavenReportException: Error while generating Javadoc: 
[ERROR] Exit code: 1 - javadoc: error - The code being documented uses modules but the packages defined in https://docs.oracle.com/javase/8/docs/api/ are in the unnamed module.
[ERROR] 
[ERROR] Command line was: /usr/local/lib/jvm/openjdk11/bin/javadoc @options
[ERROR] 
[ERROR] Refer to the generated Javadoc files in '/home/travis/build/apache/bookkeeper/bookkeeper-stats/target/apidocs' dir.
[ERROR] -> [Help 1]
[ERROR] 
```

--- a/pom.xml
+++ b/pom.xml
@@ -188,7 +188,7 @@
     <maven-deploy-plugin.version>2.7</maven-deploy-plugin.version>
     <maven-install-plugin.version>2.5.1</maven-install-plugin.version>
     <maven-jar-plugin.version>2.4</maven-jar-plugin.version>
-    <maven-javadoc-plugin.version>3.0.0</maven-javadoc-plugin.version>
+    <maven-javadoc-plugin.version>3.1.1</maven-javadoc-plugin.version>
     <maven-shade-plugin.version>3.1.0</maven-shade-plugin.version>
     <maven-source-plugin.version>2.2.1</maven-source-plugin.version>
     <maven-surefire-plugin.version>2.21.0</maven-surefire-plugin.version>
@@ -904,6 +904,7 @@
           <doctitle>BookKeeper Java API (version ${project.version})</doctitle>
           <overview>site/_site/overview/index.html</overview>
           <show>package</show>
+          <detectJavaApiLink>false</detectJavaApiLink>
         </configuration>
         <executions>
           <execution>

